### PR TITLE
container prune remove state created, configured

### PR DESCRIPTION
**Pull request by QiWang19**
podman container prune should remove containers with state Created, Configured to be compatible with docker

Signed-off-by: Qi Wang <qiwan@redhat.com>

### CODE REVIEW

**Comment by mheon** [comment]
/approve
LGTM

File: test/e2e/prune_test.go [hunk 1]
```diff
@@ -59,6 +59,18 @@ var _ = Describe("Podman prune", func() {
 		Expect(podmanTest.NumberOfContainers()).To(Equal(1))
 	})
 
+	It("podman container prune after create containers", func() {
+		stop := podmanTest.Podman([]string{"create", "--name", "test", BB})
+		stop.WaitWithDefaultTimeout()
+		Expect(stop.ExitCode()).To(Equal(0))
+
+		prune := podmanTest.Podman([]string{"container", "prune", "-f"})
```
Comment (TomSweeneyRedHat) [inline test/e2e/prune_test.go]: Can you add a container that's in the "Configured"  state to this test to make sure it's removed?

**Comment by TomSweeneyRedHat** [comment]
Other prune tests look to need adjusting.

File: test/e2e/prune_test.go (hunk 1 above)
Comment (QiWang19) [reply test/e2e/prune_test.go]: after `podman create` the container status is `Configured` https://bugzilla.redhat.com/show_bug.cgi?id=1810874#c7

Comment (TomSweeneyRedHat) [reply test/e2e/prune_test.go]: OK.  My main ask is we're now pruning containers that are in the Created and Configured state.  Can you set up at least one of each and make sure they get pruned?  Currently, it looks like we're only testing Configured  then.

Comment (QiWang19) [reply test/e2e/prune_test.go]: Do you know how to create the container has status `Created`?
```
$ podman create --name ctr alpine
bac6adb052817e1a9ea88966753b8a4c6f389e6b5116c17f33ec22475c48d302
$ podman inspect ctr --format {{.State.Status}}
configured
```
```
$ podman run --name ctr --entrypoint /error alpine
Error: container_linux.go:346: starting container process caused "exec: \"/error\": stat /error: no such file or directory": OCI runtime command not found error
c0e217d5c193  docker.io/library/alpine:latest  /bin/sh  3 seconds ago  Created         charming_booth
$ podman inspect ctr --format {{.State.Status}}
configured
```

**Comment by mheon** [comment]
`podman init` on a Configured container will make you a Created container

**Comment by QiWang19** [comment]
@mheon @TomSweeneyRedHat PTAL

**Comment by rhatdan** [comment]
@mheon @TomSweeneyRedHat @QiWang19 Is this ready to go in?

**Comment by mheon** [comment]
LGTM

**Pull request merged**